### PR TITLE
use read lock first.

### PR DESCRIPTION
**Pull request by siddontang**
@overvenus @disksing

### CODE REVIEW

**Comment by disksing** [comment]
LGTM

File: src/vec.rs [hunk 1]
```diff
@@ -61,13 +61,16 @@ impl<T: MetricVecBuilder> MetricVecCore<T> {
     pub fn get_metric_with_label_values(&self, vals: &[&str]) -> Result<T::Output> {
         let h = try!(self.hash_label_values(&vals));
 
+        if let Some(metric) = self.children.read().unwrap().get(&h).cloned() {
+            return Ok(metric);
+        }
```
Comment (overvenus) [inline src/vec.rs]: Why clone at line 64? I think we can clone at line 65.

Comment (siddontang) [inline src/vec.rs]: no big different.

Comment (siddontang) [inline src/vec.rs]: PTAL @overvenus

**Comment by overvenus** [comment]
LGTM

发自我的 iPhone

**Pull request merged**